lib/vjs-hls.js

```diff
@@ -31,6 +31,27 @@ var attachVideojsStreamrootProvider = function (window, videojs, Hls) {
 
         function _onError(event, data) {
             console.error('HLS.js error: ' + data.type + ' - fatal: ' + data.fatal + ' - ' + data.details);
+            
+            // implement simple error handling based on hls.js documentation (https://github.com/dailymotion/hls.js/blob/master/API.md#fifth-step-error-handling)
+            if (data.fatal) {
+                switch (data.type) {
+                    case Hls.ErrorTypes.NETWORK_ERROR:
+                        // try to recover network error
+                        console.info("fatal network error encountered, try to recover");
+                        _hls.startLoad();
+                        break;
+
+                    case Hls.ErrorTypes.MEDIA_ERROR:
+                        console.info("fatal media error encountered, try to recover");
+                        _hls.recoverMediaError();
```

Comment:
The doc also suggest to try to swap audio codec [here](https://github.com/dailymotion/hls.js/blob/master/API.md#hlsswapaudiocodec) to work around AAC HE / LC switching I guess.
We might want to add that here, but I'm not sure if you're supposed to call it after a first try with just calling `recoverMediaError` or not.